scripts/consensus_merge.py

```diff
@@ -0,0 +1,636 @@
+            DP_list.append(str(targ_sample.DP))
+
+        GT = stringify(GT_list, FORMAT_JOIN)
+        AD = stringify(AD_list, FORMAT_JOIN)
+        AF = stringify(AF_list, FORMAT_JOIN)
+        DP = stringify(DP_list, FORMAT_JOIN)
+
+        output_record.samples[index]['GTC'] = GT
+        output_record.samples[index]['ADC'] = AD
+        output_record.samples[index]['AFC'] = AF
+        output_record.samples[index]['DPC'] = DP
+
+        consensus_gt, gt_tag = get_gt_consensus(GT_list)
+
+#        print(consensus_gt, tuple(consensus_gt.split('/')))
+        output_record.samples[index]['GT'] = tuple([int(i) for i in consensus_gt.split('/')])
+        output_record.samples[index]['GT_STATUS'] = gt_tag
+        output_record.samples[index]['AD'] = get_ad_consensus(AD_list)
+        output_record.samples[index]['AF'] = get_af_consensus(AF_list)
+        output_record.samples[index]['DP'] = get_dp_consensus(DP_list)
+
+    chrom = single_caller_variants[0].record.chrom
+    pos = single_caller_variants[0].record.pos
+    mapq, mq0 = get_mapq(normal_cram, chrom, pos)
+
+    output_record.info['MQ'] = mapq
+    output_record.info['MQ0'] = mq0
+    output_record.info['CAL'] = ','.join([c for c in CALLER_NAMES if c.lower() in variant_lookup])
+
+    if hotspot:
+        output_record.info['Hotspot'] = True
+
+    output_record.filter.add('PASS')
+
+    output_vcf.write(output_record)
+
+def build_output_name(inpath, outbase):
+    """Builds an VCF.GZ output filename based on the input (VCF/VCF.GZ) name and given output basename
+       Args:
+           inpath (str): Path to the input VCF(.GZ) file
+           outbase (str): Given output basename string
+       Return:
+           str: Filename in the format <output_basename>.<input_nameroot>.consensus.vcf.gz
+    """
+    basename_split = os.path.split(inpath)[-1].split('.')
+    return '.'.join([outbase, basename_split[0] + '.consensus.vcf.gz'])
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--strelka2_vcf', 
+                        help='Strelka2 VCF with MNPs fixed')
+    parser.add_argument('--mutect2_vcf',
+                        help='Mutect2 VCF')
+    parser.add_argument('--lancet_vcf', 
+                        help='Lancet VCF')
+    parser.add_argument('--vardict_vcf',
+                        help='VarDict VCF')
+    parser.add_argument('--cram', help='CRAM file')
+    parser.add_argument('--output_basename',
+                        help='String to use as basename for output file')
+
+    args = parser.parse_args()
+
+    # Create VCF and CRAM pysam objects
+    strelka2_vcf = pysam.VariantFile(args.strelka2_vcf, 'r')
+    mutect2_vcf = pysam.VariantFile(args.mutect2_vcf, 'r')
+    lancet_vcf = pysam.VariantFile(args.lancet_vcf, 'r')
+    vardict_vcf = pysam.VariantFile(args.vardict_vcf, 'r')
+
+    normal_cram = pysam.AlignmentFile(args.cram, 'rc') #, reference_filename="/home/ubuntu/volume/ref/Homo_sapiens_assembly38.fasta.fai")
```

Comment:
Hmm, why only normal cram? Is the MQ info from the tumor cram not relevant?

Reply:
I was taking the example of the Strelka2 vcf in which MQ and MQ0 are INFO fields, so one value per tumor / normal pair. To me it's not clear which sample they derive this value from, or if they go to the effort of combining them. I would say that mapping quality is mostly a function of read length and reference sequence and there is probably not much information to be gained by providing separate values for the normal and tumor crams, but we can go down that road if e.g. downstream tools are expecting MQ / MQ0 FORMAT fields.